pkg/work/hub/manager.go

```diff
@@ -2,73 +2,166 @@ package hub
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 
 	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
 	clusterinformers "open-cluster-management.io/api/client/cluster/informers/externalversions"
 	workclientset "open-cluster-management.io/api/client/work/clientset/versioned"
 	workinformers "open-cluster-management.io/api/client/work/informers/externalversions"
+	workapplier "open-cluster-management.io/sdk-go/pkg/apis/work/v1/applier"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/generic/options/grpc"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/generic/options/mqtt"
+	cloudeventswork "open-cluster-management.io/sdk-go/pkg/cloudevents/work"
+	"open-cluster-management.io/sdk-go/pkg/cloudevents/work/source/codec"
 
+	"open-cluster-management.io/ocm/pkg/common/options"
 	"open-cluster-management.io/ocm/pkg/work/hub/controllers/manifestworkreplicasetcontroller"
 )
 
-// RunWorkHubManager starts the controllers on hub.
-func RunWorkHubManager(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
-	hubWorkClient, err := workclientset.NewForConfig(controllerContext.KubeConfig)
-	if err != nil {
-		return err
+const (
+	sourceID       = "mwrsctrl"
+	sourceClientID = "mwrsctrl-client"
```

Comment:
nit: Is this safe to have the same hard-code clientID for mwrs controller?
does this mean anytime, only one mwrs controller instance is working?

Reply:
maybe need to expose these with flags